src/AspNet.Security.OAuth.QQ/QQAuthenticationHandler.cs

```diff
@@ -111,15 +116,25 @@ protected override async Task<OAuthTokenResponse> ExchangeCodeAsync([NotNull] OA
                 return OAuthTokenResponse.Failed(new Exception("An error occurred while retrieving an access token."));
             }
 
-            var content = QueryHelpers.ParseQuery(await response.Content.ReadAsStringAsync());
-            var payload = await CopyPayloadAsync(content);
-
+            using var rspStream = await response.Content.ReadAsStreamAsync();
+            var payload = JsonDocument.Parse(rspStream);
             return OAuthTokenResponse.Success(payload);
         }
 
-        private async Task<string> GetUserIdentifierAsync(OAuthTokenResponse tokens)
+        private async Task<(int errorCode, string openid, string unionid)> GetUserIdentifierAsync(OAuthTokenResponse tokens)
         {
-            string address = QueryHelpers.AddQueryString(Options.UserIdentificationEndpoint, "access_token", tokens.AccessToken);
+            // Details see "https://wiki.connect.qq.com/unionid%E4%BB%8B%E7%BB%8D"
+            var queryString = new Dictionary<string, string>(3)
+            {
+                { "access_token", tokens.AccessToken },
+                { "fmt", "json" }, // Return json instead of jsonpb which is default due to historical reasons
```

Comment:
```suggestion
                ["access_token"] = tokens.AccessToken,
                ["fmt"] = "json", // Return JSON instead of jsonpb which is default due to historical reasons
```